src/workspace/Lesson/hooks.js

```diff
@@ -1,282 +1,91 @@
+            const sections = loadSectionData(...cachedLessonData.sections)
+
+            dispatch(setSections(sections))
+
+            if(sections.length !== cachedLessonData.sections.length){
+                const sectionsNotInChache = cachedLessonData.sections.filter(secId => !sections.find(section => section.id === secId))
+                dispatch(fetchSection(...sectionsNotInChache))
             }
-            dispatch({ type: "BOOTSTRAP", payload: lesson })
-        } catch (err) {
-            dispatch({ type: "ERROR" })
-            dispatch({
-                type: "BOOTSTRAP",
-                payload: {
-                    id: new Date().getTime(),
-                    sections: [
-                        {
-                            id:
-                                new Date().getTime() +
-                                "" +
-                                Math.floor(Math.random() * 100),
-                            docValue: null,
-                            visible: true,
-                        },
-                    ],
-                    changed: new Set(),
-                },
-            })
+
+
+		} else {
+           await dispatch(fetchLessonWithSections(id, courseId))
         }
+
         requestAnimationFrame(() => {
             dispatch({ type: "BOOTSTRAP_FINISH" })
         })
     }
 
-    async function fetchCourse() {
-        try {
-            // const courseId = window.location.pathname.split("/")[2]
-            const course = await serverApi.get(`/courses/${courseId}`)
-            dispatch({ type: "SET_COURSE", payload: course })
-        } catch (err) {
-            console.log(err)
-        }
-    }
-
     async function registerHandler(){
-        
-        const dispatchLessonUpdate = (data) => {
-            
-            dispatch({
-                type: "LESSON_UPDATED",
-                payload: data
-            })
-        }
+        const dispatchLessonUpdate = (data) =>
+            dispatch(lessonWasUpdated(data))
+
         editorWS.on('course/:courseId/lessons patched', dispatchLessonUpdate)
         editorWS.on('course/:courseId/lessons updated', dispatchLessonUpdate)
 
-        const dispatchSectionUpdate = (data) => {
-            dispatch({
-                type: "SECTION_UPDATED",
-                payload: data
-            })
-        }
+        const dispatchSectionUpdate = (data) =>
+            dispatch(sectionWasUpdated(data._id, data))
 
         editorWS.on('lesson/:lessonId/sections patched', dispatchSectionUpdate)
         editorWS.on('lesson/:lessonId/sections updated', dispatchSectionUpdate)
 
 
-        const dispatchSectionDiff = (data) => {
-            dispatch({
-                type: "SECTION_DOCVALUE_DIFF",
-                payload: {
-                    sectionId: data._id,
-                    diff: data.diff
-                }
-            })
-        }
+        const dispatchSectionDiff = (data) => dispatch(mergeSerloDiff(data._id, data.diff))
```

Comment:
data._id sieht hier irgendwie falsch aus

Reply:
sind die daten die vom server kommen, ist also richtig soweit, kann data aber auch secitons umbennen wenn das der eigentliche anstoß ist :D

Reply:
Aber dann sollten wir vllt tatsächlich die anderen sachen mergen, das hab ich schon teilweise umgeschrieben und extiert so nicht mehr

Reply:
is das im aktuellen statium ok?